rdkit/Chem/AllChem.py

```diff
@@ -499,34 +542,52 @@ def GenerateStereoisomers(m,options=StereoEnumerationOptions(),verbose=False):
 
     """
     tm = Mol(m)
-    if options.onlyUnassigned:
-        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True) if y=='?']
-    else:
-        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True)]
-    nCenters = len(possibleCenters)
+    flippers = _getFlippers(tm, options)
+    nCenters = len(flippers)
     if not nCenters:
         yield tm
         return
-    if nCenters>options.maxNumCenters:
-        raise ValueError("nCenters (%d) larger than maxNumCenters (%d)"%(nCenters,options.maxNumCenters))
-    bitflag = (1<<nCenters)-1
-    while bitflag>=0:
-        tm = Mol(m)
+
+    # this sillyness is to deal with Python 3 getting rid of xrange
+    # and range slurping up 2**N memory in Python 2
+    range_func = range
```

Comment:
We tend to use six.moves.range

Reply:
Done, that's what I get for only looking for APIs with ipython tab completion